Simplify the RFC process (#8)
getsentry/rfcs: text/0001-workflow.md

* `withdrawn`: the RFC was withdrawn.  Typically such RFCs are not visible in the repository as the corresponding PRs are not merged unless they are withdrawn after accepted.
* `replaced`: the RFC was later replaced by another RFC.

# Workflow

> 1. Create a new RFC document as a copy of `0000-template.md`. Name it `XXXX-short-description.md` and commit it directly to the `main` branch. Anyone can do this at any time.

Author: ```suggestion
1. Create a new RFC document as a copy of `0000-template.md`. Name it `XXXX-short-description.md` and commit it directly to the `main` branch. Anyone can do this at any time. Publishing a new doc to `main` is the "request" in RFC.
```

> 1. Create a new RFC document as a copy of `0000-template.md`. Name it `XXXX-short-description.md` and commit it directly to the `main` branch. Anyone can do this at any time. Publishing a new doc to `main` is the "request" in RFC.

Author: ```suggestion
1. Create a new RFC document as a copy of `0000-template.md`. Name it `XXXX-short-description.md` (number it one greater than the current max) and commit it directly to the `main` branch. Anyone can do this at any time. Publishing a new doc to `main` is the "request" in RFC.
```

1. Modify your RFC directly on `main` or through PRs as you see fit, except that ...

> 1. For non-informational RFCs, updating the RFC status to `approved` must happen in a PR, with relevant parties signing off.

Reviewer: For RFCs that are made on main, how would the 'comments' be done? One strength of having the entire RFC show up as a pull request is the workflow that github provides around providing feedback and incorporating that feedback.

Author: I guess the mental model is: publishing on main is the "request", and subsequent PRs are the "comments." How about this?

```suggestion
1. Modify your RFC directly on `main` until other people care about it, then switch to PRs to track comments and changes.
1. For non-informational RFCs, updating the RFC status to `approved` _must_ happen in a PR, with relevant parties signing off.
```

Author: > subsequent PRs

... or Issues, or Discussions, or Discord, or Discord Forums, or ... 😭

Author: Right now PRs that add a new RFC are requests for comments about requests for comments. I propose we unnest. :)
